src/bulk/common.ts

```diff
@@ -328,34 +329,61 @@ export class BulkWriteResult {
   }
 }
 
+/** @internal */
+export interface WriteConcernErrorData {
+  code: number;
+  errmsg: string;
+  errInfo?: Document;
+}
+
 /**
  * An error representing a failure by the server to apply the requested write concern to the bulk operation.
  * @public
  * @category Error
  */
 export class WriteConcernError {
-  err: MongoServerError;
+  /** @internal */
+  [kServerError]: WriteConcernErrorData;
 
-  constructor(err: MongoServerError) {
-    this.err = err;
+  /** @internal */
+  constructor(error: WriteConcernErrorData) {
+    this[kServerError] = error;
   }
 
   /** Write concern error code. */
   get code(): number | undefined {
-    return this.err.code;
+    return this[kServerError].code;
   }
 
   /** Write concern error message. */
-  get errmsg(): string {
-    return this.err.errmsg;
+  get errmsg(): string | undefined {
+    return this[kServerError].errmsg;
   }
 
-  toJSON(): { code?: number; errmsg: string } {
-    return { code: this.err.code, errmsg: this.err.errmsg };
+  /** Write concern error info. */
+  get errInfo(): Document | undefined {
+    return this[kServerError].errInfo;
+  }
+
+  /**
+   * @internal
```

Comment:
does it make sense to mark this internal if it's explicitly intended as a backcompat option for users?

Reply:
I suppose my thinking was it avoids breaking runtime code but intentionally makes it inaccessible to TS but I realize that could be nearly equal in breakage. 

lets go with making this interface public and not introduce all these specialized exceptions. Sorry @andymina my advice steered you here!!

Reply:
No worries!